Solution_1.1

yaroslavsysoiev opened the pull request.

Dimosphen1 commented on app/main.py:
```diff
@@ -1 +1,16 @@
-# write your code here
+from typing import List
+from app.players.player import Player
+from app.players.elves.elf import Elf
+from app.players.dwarves.dwarf import Dwarf
```
Separate different groups of imports with one blank line according to PEP-8.

Dimosphen1 commented on app/main.py:
```diff
@@ -1 +1,16 @@
+    return sum(player.get_rating() for player in team)
+
+
+def elves_concert(elves: List[Elf]) -> List[None]:
+    return [elf.play_elf_song() for elf in elves]
```
There is no need to return a list of `None` in these methods, omit using the `return` statement and change your return type annotation accordingly.

Dimosphen1 commented on app/players/dwarves/dwarf.py:
```diff
@@ -0,0 +1,12 @@
+from abc import ABC
+from app.players.player import Player
```
Separate different groups of imports with one blank line according to PEP-8.

Dimosphen1 commented on app/players/elves/elf.py:
```diff
@@ -0,0 +1,13 @@
+from abc import ABC
+from app.players.player import Player
```
Separate different groups of imports with one blank line according to PEP-8.

Dimosphen1 commented on tests/test_main.py:
```diff
@@ -327,9 +327,9 @@ def test_elves_concert(elves, songs):
+                    DwarfWarrior(nickname="Dwarf2", _favourite_dish="French Fries", hummer_level=3),
                 ],
                 (
                         "Thiddeal is eating French Fries\n"
```
Revert your changes in tests and make appropriate changes in the signature of the `__init__` of the underlying classes. You should take an argument without `_` and later write its value to the instance attribute like `self._...`.

Dimosphen1 reviewed (changes requested):
Several changes were requested.

Dimosphen1 commented on tests/test_main.py:
```diff
@@ -316,14 +316,14 @@ def test_elves_concert(elves, songs):
+                )
         ),
         (
                 [
```
Revert any changes done to the `tests/test_main.py` file. You could copy the content from the base repository, paste it to the current file, and commit changes.

yaroslavsysoiev replied:
Well, actually, that what I did before :) 
I went to mate-academy / py-elves-and-dwarves -> Tests -> test_main.py and copied the content out of there and pasted to my test/main.py.
I will do it one more time. If I misunderstood, please inform me.

Dimosphen1 commented on __init__.py:
There is no need for this `__init__.py` file, remove it.

Dimosphen1 commented on app/players/elves/druid.py:
```diff
@@ -0,0 +1,20 @@
+
+class Druid(Elf):
+    def __init__(
+            self,
+            nickname: str,
+            musical_instrument: str,
+            favourite_spell: str
```
Your signature for all classes `__init__` seems over-indented. Decrease indentation here and in other classes as well.

Dimosphen1 reviewed (changes requested):
Several changes were requested.

yaroslavsysoiev commented:
1. I wanted to add unused init.py files to gitignore. So I wrote echo > .gitignore and added them. When I was re-reading the cooments and decided to delete them. So I deleted the init.py files and gitignore as I had no need in it any more. 
2.  Okay, I copy-pasted the tests from mate-academy repo one more time, They must be the same as before.

yaroslavsysoiev commented:
Okay, so I used echo > .gitignore to create gitignore. It seems to me the only way to rool it back. 
I failed to run 'git push origin develop' because 'Everything up-to-date' occured. 
So I changed the code a little bit and ran 'pytest' and 'flake8 app' one more time. 
And then successfully pushed.

Dimosphen1 commented on .gitignore:
Do not delete the `.gitignore` file, leave it in place with content from the [base repository](https://github.com/mate-academy/py-elves-and-dwarves/blob/master/.gitignore).

Dimosphen1 commented on __init__.py:
There is no need for this `__init__.py` at the root level. Remove it from the pull request.

Dimosphen1 reviewed (changes requested):
Several changes were requested.

pavlejviki commented on app/main.py:
```diff
@@ -1 +1,17 @@
+
+
+def elves_concert(elves: List[Elf]) -> None:
+    [elf.play_elf_song() for elf in elves]
```
there is no need in the list comprehension as you don't need to store the results of the function. a simple loop is enough

pavlejviki commented on app/main.py:
```diff
@@ -1 +1,17 @@
+
+
+def feast_of_the_dwarves(dwarves: List[Dwarf]) -> None:
+    [dwarf.eat_favourite_dish() for dwarf in dwarves]
```
see comment above

Polyakiv-Andrey reviewed (approved):
Good job!
